### PR TITLE
Feature/iotsdk 226 json decoupling

### PR DESCRIPTION
**Pull request by vputys**

### CODE REVIEW

File: .gitignore
Comment (pywtk) [inline .gitignore:1]: Any reason for this change?

File: meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/files/cmke-src/CMakeLists.txt
Comment (pywtk) [inline meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/files/cmke-src/CMakeLists.txt:1]: I see that the file structure is this:
cmke-src/
├── CMakeLists.txt
├── components
│   └── json_parser
│       ├── CMakeLists.txt
│       ├── inc
│       │   └── json_parser.h
│       └── json_parser.c
├── config
│   └── app_config.h
└── main.c

& the CMakeLists.txt files support this.

It seems to me the same resutls can be achieved with a structure like this:
cmke-src/
├── CMakeLists.txt
├── config
│   └── app_config.h
├── json_parser.c
├── json_parser.h
└── main.c

& a single change to the one CMakeLists.txt file:

--- a/meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/files/cmke-src/CMakeLists.txt
+++ b/meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/files/cmke-src/CMakeLists.txt
@@ -11,5 +11,5 @@ include(GNUInstallDirs)
 
 include_directories(config PRIVATE ${IOTC_C_LIB_INCLUDE_DIR} ${IOTC_C_SDK_INCLUDE_DIR})
 add_executable(telemetry-demo main.c)
-
+target_sources(telemetry-demo PUBLIC json_parser.c)
 target_link_libraries(telemetry-demo iotc-c-generic-sdk)

I'm worried I'm missing something important about the proposed approach, wee chat to discuss?

File: meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/files/cmke-src/components/json_parser/CMakeLists.txt
Comment (pywtk) [inline meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/files/cmke-src/components/json_parser/CMakeLists.txt:1]: Comment above exploring the requirement for multiple CMakeList.txt files

File: meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/files/eg-private-repo-data/.gitignore
Comment (pywtk) [inline meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/files/eg-private-repo-data/.gitignore:1]: Comment above re more gitignores?

File: meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/files/eg-private-repo-data/config-symmtrcKy.json
Comment (pywtk) [inline meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/files/eg-private-repo-data/config-symmtrcKy.json:1]: why delete these files?

File: meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/telemetry-demo_0.1.bb
Comment (pywtk) [inline meta-myExampleIotconnectLayer/recipes-apps/telemetry-demo/telemetry-demo_0.1.bb:1]: Comment above re extra CMake changes & presumably the implied required change to the recipe here

**Review by pywtk — CHANGES_REQUESTED** [review]
comments regard CMake changes & deleted files`

File: .gitignore
Comment (vputys) [reply .gitignore:1]: if I understood you right - you wanted eg-private-repo-data folder's contents to not appear on git?
I noticed that it wasn't the case and changed .gitignores then (includes the folder, but not the contents)

**Review by pywtk — APPROVED** [review]

**Review by pywtk — APPROVED** [review]
looks ok

**Pull request merged**